src/models/Movie/parseMovieData.ts

```diff
@@ -0,0 +1,15 @@
+import { MovieContentfulEntry, MovieModel } from './type';
+import { GenreContentfulEntry } from '../Genre/GenreModel';
+
+export default function parseMovieData(entry: MovieContentfulEntry): MovieModel {
+	const { sys, fields } = entry;
+	const movie = {
+		...fields,
+		id: sys.id,
+		genres: ((fields.genres as Array<GenreContentfulEntry>) || [])?.map(
+			(genre) => genre.fields?.title || ''
+		),
+	};
```

Comment:
This is a bit of a smell to me because of the `as MovieModel` below. For parsing functions like this one, explicitness would probably be better so something like:

```
const movie: MovieModel = {
  id: sys.id,
  genres: ((fields.genres as Array<GenreContentfulEntry>) || [])?.map(
			(genre) => genre.fields?.title || ''
		),
  title: fields.title,
  // etc
}
```

This ensure the shape is guaranteed to match rather than possibly having additional fields that snuck through that you didn't want. Alternatively, you could just filter out fields that aren't in the type definition. Take your pick but you want to be explicit here.